src/Symfony/Component/Validator/Constraints/UniqueValidator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Yevgeniy Zholkevskiy <zhenya.zholkevskiy@gmail.com>
+ */
+class UniqueValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Unique) {
+            throw new UnexpectedTypeException($constraint, Unique::class);
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        if (!is_array($value) && !$value instanceof \IteratorAggregate) {
```

Comment:
I've seen @ostrolucky 's https://github.com/symfony/symfony/pull/26555#discussion_r174968790, but actually, we don't need `ArrayAccess` here, right? 
So `!is_array($value) && !$value instanceof \Traversable` would be enough.

Reply:
I thought so too until recently. Apparently this is considered too unsafe, see https://github.com/symfony/symfony/pull/24577#issuecomment-358107104 and https://github.com/symfony/symfony/pull/25506

Reply:
So, IIUC that's an issue we already have with `AllValidator` then?

Was it already raised somewhere and do we need to add a deprecation to warn about this issue if `$value` isn't implementing `IteratorAggregate`?

Reply:
Dunno, what do you say @nicolas-grekas? I guess there should be an RFC to decide how to move forward in consistent way and let others to speak their mind about this.

Reply:
I'd keep it as it is.